src/command.rs

```diff
@@ -49,6 +51,95 @@ macro_rules! delegate {
     }};
 }
 
+/// If a command is `OverSsh` then it can be executed over an SSH session.
+/// Primarily a way to allow `std::process::Command` to be turned directly into an `openssh::Command`.
+pub trait OverSsh {
+    /// Given a session, return a command that can be executed over that session.
+    ///
+    /// **Note**: The command to be executed on the remote machine does not include
+    /// any environment variables or the current directory. They must be
+    /// set explicitly, if desired.
+    ///
+    /// # Example
+    /// Consider the implementation of `OverSsh` for `std::process::Command`:
+    ///
+    /// ```no_run
+    /// # #[tokio::main(flavor = "current_thread")]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     use std::process::Command;
+    ///     use openssh::{Session, KnownHosts, OverSsh};
+    ///
+    ///     let session = Session::connect_mux("me@ssh.example.com", KnownHosts::Strict).await?;
+    ///     let ls =
+    ///         Command::new("ls")
+    ///         .arg("-l")
+    ///         .arg("-a")
+    ///         .arg("-h")
+    ///         .over_session(&session)
+    ///         .output()
+    ///         .await?;
+    ///
+    ///     assert!(String::from_utf8(ls.stdout).unwrap().contains("total"));
+    /// #   Ok(())
+    /// }
+    ///
+    /// ```
+    fn over_session<'session>(&self, session: &'session Session) -> crate::Command<'session>;
+}
+
+impl OverSsh for std::process::Command {
+    fn over_session<'session>(&self, session: &'session Session) -> Command<'session> {
+        let program_escaped: Cow<'_, OsStr> = escape(self.get_program());
+        let mut command = session.raw_command(program_escaped);
+
+        let args = self.get_args().map(escape);
+        command.raw_args(args);
```

Comment:
Why use `escape` + `raw_args` rather than just calling `args` directly and have that do the escaping for us?

Reply:
Same above for `escape` + `raw_command`

Reply:
Because command/args only takes str and here we have OsStr.

Once this PR is merged, I plan to use escape introduced here for OsStr inside command/args so that they can take OsStr.